pkg/cloudprovider/providers/vsphere/nodemanager.go

```diff
@@ -47,17 +50,25 @@ type NodeManager struct {
 	registeredNodes map[string]*v1.Node
 	//CredentialsManager
 	credentialManager *SecretCredentialManager
+	// Maps node name to zone information
+	zoneInfoMap map[string]*cloudprovider.Zone
```

Comment:
Since we already have nodeInfo map that caches vSphere info related to the node VM, I would suggest to move the zonal information in the same map. This way there is no need to maintain a separate map (zoneInfoMap) and a lock(zoneInfoLock).

Reply:
Thanks for the suggestion. Fixed.